### DISCUSSION TITLE
whether the request for products come, it will pass through the order as below in one node moleculer project?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [moleculerjs/moleculer#1198] [Q&A] [answered]
When I see this architecture diagram:
![image](https://user-images.githubusercontent.com/12727582/226275357-cfabe51e-cff0-4a83-9716-af9ca4b4454e.png)

now in my moleculer demo (only one node):
```
./
├── mixins
│   └── db.mixin.js
├── moleculer.config.js
├── package-lock.json
├── package.json
└── services
    ├── api.service.js
    ├── greeter.service.js
    ├── lab.service.js
    └── products.service.js
```

I want to know whether the request for products come, it will pass through the order as below?
```
services/api.service.js -> broker -> services/products.service.js
```

**Comment by icebob** [comment, marked as answer]
If you run the API service and products service on different nodes, broker will transfer the request via transporter. If both run on the same node, broker will call it directly, in-process.
